audformat/core/table.py

```diff
@@ -446,43 +448,47 @@ def _get_by_index(
         # Returns `df, df_is_copy`
         raise NotImplementedError()
 
-    def _index_levels_and_converters(self) -> typing.Tuple[
-            typing.Sequence[str],
-            typing.Dict[str, typing.Callable],
-    ]:  # pragma: no cover
-        # Executed when loading table from CSV file.
-        # Returns: `index_levels, index_converters`
-        raise NotImplementedError()
-
     def _load_csv(self, path: str):
 
-        usecols = []
-        dtypes = {}
         schemes = self.db.schemes
+        converters = {}
+        dtypes = {}
 
-        # index columns
+        if hasattr(self, 'type'):
+            # filewise or segmented table
+            dtypes[define.IndexField.FILE] = define.DataType.STRING
+            if self.type == define.IndexType.SEGMENTED:
+                dtypes[define.IndexField.START] = define.DataType.TIME
+                dtypes[define.IndexField.END] = define.DataType.TIME
+        else:
+            # misc table
+            dtypes = self.levels
 
-        levels, converters = self._index_levels_and_converters()
-        usecols.extend(levels)
+        # index columns
+        dtypes = {
+            level: to_pandas_dtype(dtype)
+            for level, dtype in dtypes.items()
+        }
+        levels = list(dtypes)
 
         # other columns
-
         for column_id, column in self.columns.items():
-            usecols.append(column_id)
             if column.scheme_id is not None:
-                dtype = schemes[column.scheme_id].to_pandas_dtype()
-                # use converter if column contains dates or timestamps
-                if dtype == 'datetime64[ns]':
-                    converters[column_id] = lambda x: pd.to_datetime(x)
-                elif dtype == 'timedelta64[ns]':
-                    converters[column_id] = lambda x: pd.to_timedelta(x)
-                else:
-                    dtypes[column_id] = dtype
+                dtypes[column_id] = schemes[column.scheme_id].to_pandas_dtype()
             else:
                 dtypes[column_id] = 'str'
 
-        # read csv
+        # all columns
+        usecols = list(dtypes)
+
+        # extend dtype with converter for dates or timestamps
+        for column_id, dtype in dtypes.items():
+            if dtype == 'datetime64[ns]':
+                converters[column_id] = lambda x: pd.to_datetime(x)
```

Comment:
I did this before, the problem is that it will then not work for an empty index. This also seems to be a bug in `pandas` to me.
I did not see the warning message any longer at my side, so I was hoping it is not there.

> only the converter will be used

This is simply wrong as you get a different result for an empty table when specifying `dtypes` in addition.

Would be great if we could suppress the warning somehow.

Reply:
I think this is indeed the easiest solution.

I switched to checking for an empty dataframe and adjust the dtype accordingly now.
I need to add a few more tests to cover all possible cases for this in the tests.

Could you maybe check if the warning message goes away already?

Reply:
Can confirm the warning is gone.

Reply:
It was not completely trivial, but we now have also tests for `dtype` of `MultiIndex` based index and for columns.